### DISCUSSION TITLE
How to update if a new version is released?

### DISCUSSION THREAD

**Discussion by Zuendapp** [keatontaylor/alexa-actions#227] [Q&A] [answered]
Hi :),

it's a awesome code, thank you very much :)! I use it every day.

My question is how to update the skill if there is any important change from your code?
Is there any way to paritally change that, or only with setting up a new skill / lambda function?

Thank you very much!

BR

**Comment by DEADSEC-SECURITY** [comment]
This is a topic for discussions and not for issues.

**Comment by DEADSEC-SECURITY** [comment, marked as answer]
Anyway, you always have to rebuild the skill with the new code.

**Reply by Zuendapp** [reply]
Thank you!

:)

**Comment by DEADSEC-SECURITY** [comment]
The maintained update/rollback instructions are available in PR #282: https://github.com/keatontaylor/alexa-actions/pull/282 . docs/troubleshooting.md separates Alexa-hosted source/requirements builds from self-managed Lambda ZIP/runtime updates, preserves local settings and customized models, explains HA wrapper/reload changes, and lists simulator checks plus rollback.

Please try the development update steps for your hosting type and report any missing/incorrect step. This is awaiting merge/release, and the existing release is unchanged. No physical Echo is needed for the offline suite or the Alexa simulator portions.
